File: binding/src/libqprogressdialog.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qprogressdialog.html
pub const qprogressdialog = struct {

    /// New constructs a new QProgressDialog object.
    pub fn New(parent: ?*anyopaque) QtC.QProgressDialog {
        return qtc.QProgressDialog_new(@ptrCast(parent));
    }


    /// New2 constructs a new QProgressDialog object.
    pub fn New2() QtC.QProgressDialog {
        return qtc.QProgressDialog_new2();
    }


    /// New3 constructs a new QProgressDialog object.
    pub fn New3(labelText: []const u8, cancelButtonText: []const u8, minimum: i32, maximum: i32) QtC.QProgressDialog {
        const labelText_str = qtc.libqt_string{
    .len = labelText.len,
    .data = labelText.ptr,
};
const cancelButtonText_str = qtc.libqt_string{
    .len = cancelButtonText.len,
    .data = cancelButtonText.ptr,
};
return qtc.QProgressDialog_new3(labelText_str, cancelButtonText_str, minimum, maximum);
    }


    /// New4 constructs a new QProgressDialog object.
    pub fn New4(parent: ?*anyopaque, flags: i64) QtC.QProgressDialog {
        return qtc.QProgressDialog_new4(@ptrCast(parent), @intCast(flags));
    }


    /// New5 constructs a new QProgressDialog object.
    pub fn New5(labelText: []const u8, cancelButtonText: []const u8, minimum: i32, maximum: i32, parent: ?*anyopaque) QtC.QProgressDialog {
        const labelText_str = qtc.libqt_string{
    .len = labelText.len,
    .data = labelText.ptr,
};
const cancelButtonText_str = qtc.libqt_string{
    .len = cancelButtonText.len,
    .data = cancelButtonText.ptr,
};
return qtc.QProgressDialog_new5(labelText_str, cancelButtonText_str, minimum, maximum, @ptrCast(parent));
    }


    /// New6 constructs a new QProgressDialog object.
    pub fn New6(labelText: []const u8, cancelButtonText: []const u8, minimum: i32, maximum: i32, parent: ?*anyopaque, flags: i64) QtC.QProgressDialog {
        const labelText_str = qtc.libqt_string{
    .len = labelText.len,
    .data = labelText.ptr,
};
const cancelButtonText_str = qtc.libqt_string{
    .len = cancelButtonText.len,
    .data = cancelButtonText.ptr,
};
return qtc.QProgressDialog_new6(labelText_str, cancelButtonText_str, minimum, maximum, @ptrCast(parent), @intCast(flags));
    }


    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QProgressDialog_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprogressdialog.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetLabel(self: ?*anyopaque, label: ?*anyopaque) void {
        qtc.QProgressDialog_SetLabel(@ptrCast(self), @ptrCast(label));
    }

    pub fn SetCancelButton(self: ?*anyopaque, button: ?*anyopaque) void {
        qtc.QProgressDialog_SetCancelButton(@ptrCast(self), @ptrCast(button));
    }

    pub fn SetBar(self: ?*anyopaque, bar: ?*anyopaque) void {
        qtc.QProgressDialog_SetBar(@ptrCast(self), @ptrCast(bar));
    }

    pub fn WasCanceled(self: ?*anyopaque, ) bool {
        return qtc.QProgressDialog_WasCanceled(@ptrCast(self));
    }

    pub fn Minimum(self: ?*anyopaque, ) i32 {
        return qtc.QProgressDialog_Minimum(@ptrCast(self));
    }

    pub fn Maximum(self: ?*anyopaque, ) i32 {
        return qtc.QProgressDialog_Maximum(@ptrCast(self));
    }

    pub fn Value(self: ?*anyopaque, ) i32 {
        return qtc.QProgressDialog_Value(@ptrCast(self));
    }

    pub fn SizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QProgressDialog_SizeHint(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnSizeHint(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) QtC.QSize) void {
        qtc.QProgressDialog_OnSizeHint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseSizeHint(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QProgressDialog_QBaseSizeHint(@ptrCast(self));
    }

    pub fn LabelText(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QProgressDialog_LabelText(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprogressdialog.LabelText: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn MinimumDuration(self: ?*anyopaque, ) i32 {
        return qtc.QProgressDialog_MinimumDuration(@ptrCast(self));
    }

    pub fn SetAutoReset(self: ?*anyopaque, reset: bool) void {
        qtc.QProgressDialog_SetAutoReset(@ptrCast(self), reset);
    }

    pub fn AutoReset(self: ?*anyopaque, ) bool {
        return qtc.QProgressDialog_AutoReset(@ptrCast(self));
    }

    pub fn SetAutoClose(self: ?*anyopaque, close: bool) void {
        qtc.QProgressDialog_SetAutoClose(@ptrCast(self), close);
    }

    pub fn AutoClose(self: ?*anyopaque, ) bool {
        return qtc.QProgressDialog_AutoClose(@ptrCast(self));
    }

    pub fn Open(self: ?*anyopaque, receiver: ?*anyopaque, member: []const u8) void {
        qtc.QProgressDialog_Open(@ptrCast(self), @ptrCast(receiver), @ptrCast(member));
    }

    pub fn Cancel(self: ?*anyopaque, ) void {
        qtc.QProgressDialog_Cancel(@ptrCast(self));
    }

    pub fn Reset(self: ?*anyopaque, ) void {
        qtc.QProgressDialog_Reset(@ptrCast(self));
    }

    pub fn SetMaximum(self: ?*anyopaque, maximum: i32) void {
        qtc.QProgressDialog_SetMaximum(@ptrCast(self), maximum);
    }

    pub fn SetMinimum(self: ?*anyopaque, minimum: i32) void {
        qtc.QProgressDialog_SetMinimum(@ptrCast(self), minimum);
    }

    pub fn SetRange(self: ?*anyopaque, minimum: i32, maximum: i32) void {
        qtc.QProgressDialog_SetRange(@ptrCast(self), minimum, maximum);
    }

    pub fn SetValue(self: ?*anyopaque, progress: i32) void {
        qtc.QProgressDialog_SetValue(@ptrCast(self), progress);
    }

    pub fn SetLabelText(self: ?*anyopaque, text: []const u8) void {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
qtc.QProgressDialog_SetLabelText(@ptrCast(self), text_str);
    }

    pub fn SetCancelButtonText(self: ?*anyopaque, text: []const u8) void {
        const text_str = qtc.libqt_string{
    .len = text.len,
    .data = text.ptr,
};
qtc.QProgressDialog_SetCancelButtonText(@ptrCast(self), text_str);
    }

    pub fn SetMinimumDuration(self: ?*anyopaque, ms: i32) void {
        qtc.QProgressDialog_SetMinimumDuration(@ptrCast(self), ms);
    }

    pub fn Canceled(self: ?*anyopaque, ) void {
        qtc.QProgressDialog_Canceled(@ptrCast(self));
    }

    pub fn OnCanceled(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QProgressDialog_Connect_Canceled(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ResizeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_ResizeEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnResizeEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QProgressDialog_OnResizeEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseResizeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_QBaseResizeEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn CloseEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_CloseEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnCloseEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QProgressDialog_OnCloseEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCloseEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_QBaseCloseEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ChangeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_ChangeEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnChangeEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QProgressDialog_OnChangeEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseChangeEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_QBaseChangeEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ShowEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_ShowEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnShowEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QProgressDialog_OnShowEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseShowEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QProgressDialog_QBaseShowEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ForceShow(self: ?*anyopaque, ) void {
        qtc.QProgressDialog_ForceShow(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnForceShow(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QProgressDialog_OnForceShow(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseForceShow(self: ?*anyopaque, ) void {
        qtc.QProgressDialog_QBaseForceShow(@ptrCast(self));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QProgressDialog_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprogressdialog.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QProgressDialog_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qprogressdialog.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QProgressDialog_Delete(@ptrCast(self));
    }
};

File: binding/src/libqprogressdialog.cpp
#include <QCloseEvent>
#include <QDialog>
#include <QEvent>
#include <QLabel>
#include <QObject>
#include <QProgressBar>
#include <QProgressDialog>
#include <QPushButton>
#include <QResizeEvent>
#include <QShowEvent>
#include <QSize>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QWidget>
#include <qprogressdialog.h>
#include "libqprogressdialog.h"
#include "libqprogressdialog.hxx"

QProgressDialog* QProgressDialog_new(QWidget* parent) {
	 return new VirtualQProgressDialog(parent);
}

QProgressDialog* QProgressDialog_new2() {
	 return new VirtualQProgressDialog();
}

QProgressDialog* QProgressDialog_new3(const libqt_string labelText, const libqt_string cancelButtonText, int minimum, int maximum) {
	 return new VirtualQProgressDialog(QString::fromUtf8(labelText.data, labelText.len), QString::fromUtf8(cancelButtonText.data, cancelButtonText.len), minimum, maximum);
}

QProgressDialog* QProgressDialog_new4(QWidget* parent, int flags) {
	 return new VirtualQProgressDialog(parent, static_cast<QFlags<Qt::WindowType>>(flags));
}

QProgressDialog* QProgressDialog_new5(const libqt_string labelText, const libqt_string cancelButtonText, int minimum, int maximum, QWidget* parent) {
	 return new VirtualQProgressDialog(QString::fromUtf8(labelText.data, labelText.len), QString::fromUtf8(cancelButtonText.data, cancelButtonText.len), minimum, maximum, parent);
}

QProgressDialog* QProgressDialog_new6(const libqt_string labelText, const libqt_string cancelButtonText, int minimum, int maximum, QWidget* parent, int flags) {
	 return new VirtualQProgressDialog(QString::fromUtf8(labelText.data, labelText.len), QString::fromUtf8(cancelButtonText.data, cancelButtonText.len), minimum, maximum, parent, static_cast<QFlags<Qt::WindowType>>(flags));
}

libqt_string QProgressDialog_Tr(const char* s) {
	QString _ret = QProgressDialog::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QProgressDialog_SetLabel(QProgressDialog* self, QLabel* label) {
	self->setLabel(label);
}

void QProgressDialog_SetCancelButton(QProgressDialog* self, QPushButton* button) {
	self->setCancelButton(button);
}

void QProgressDialog_SetBar(QProgressDialog* self, QProgressBar* bar) {
	self->setBar(bar);
}

bool QProgressDialog_WasCanceled(const QProgressDialog* self) {
	return self->wasCanceled();
}

int QProgressDialog_Minimum(const QProgressDialog* self) {
	return self->minimum();
}

int QProgressDialog_Maximum(const QProgressDialog* self) {
	return self->maximum();
}

int QProgressDialog_Value(const QProgressDialog* self) {
	return self->value();
}

QSize* QProgressDialog_SizeHint(const QProgressDialog* self) {
	return new QSize(self->sizeHint());
}

libqt_string QProgressDialog_LabelText(const QProgressDialog* self) {
	QString _ret = self->labelText();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QProgressDialog_MinimumDuration(const QProgressDialog* self) {
	return self->minimumDuration();
}

void QProgressDialog_SetAutoReset(QProgressDialog* self, bool reset) {
	self->setAutoReset(reset);
}

bool QProgressDialog_AutoReset(const QProgressDialog* self) {
	return self->autoReset();
}

void QProgressDialog_SetAutoClose(QProgressDialog* self, bool close) {
	self->setAutoClose(close);
}

bool QProgressDialog_AutoClose(const QProgressDialog* self) {
	return self->autoClose();
}

void QProgressDialog_Open(QProgressDialog* self, QObject* receiver, const char* member) {
	self->open(receiver, member);
}

void QProgressDialog_Cancel(QProgressDialog* self) {
	self->cancel();
}

void QProgressDialog_Reset(QProgressDialog* self) {
	self->reset();
}

void QProgressDialog_SetMaximum(QProgressDialog* self, int maximum) {
	self->setMaximum(maximum);
}

void QProgressDialog_SetMinimum(QProgressDialog* self, int minimum) {
	self->setMinimum(minimum);
}

void QProgressDialog_SetRange(QProgressDialog* self, int minimum, int maximum) {
	self->setRange(minimum, maximum);
}

void QProgressDialog_SetValue(QProgressDialog* self, int progress) {
	self->setValue(progress);
}

void QProgressDialog_SetLabelText(QProgressDialog* self, const libqt_string text) {
	self->setLabelText(QString::fromUtf8(text.data, text.len));
}

void QProgressDialog_SetCancelButtonText(QProgressDialog* self, const libqt_string text) {
	self->setCancelButtonText(QString::fromUtf8(text.data, text.len));
}

void QProgressDialog_SetMinimumDuration(QProgressDialog* self, int ms) {
	self->setMinimumDuration(ms);
}

void QProgressDialog_Canceled(QProgressDialog* self) {
	self->canceled();
}

void QProgressDialog_Connect_Canceled(QProgressDialog* self, intptr_t slot) {
    void (*slotFunc)(QProgressDialog*) = reinterpret_cast<void (*)(QProgressDialog*)>(slot);
    QProgressDialog::connect(self, &QProgressDialog::canceled, [self, slotFunc]() {
	slotFunc(self);
    });
}

libqt_string QProgressDialog_Tr2(const char* s, const char* c) {
	QString _ret = QProgressDialog::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QProgressDialog_Tr3(const char* s, const char* c, int n) {
	QString _ret = QProgressDialog::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

// Base class handler implementation
QSize* QProgressDialog_QBaseSizeHint(const QProgressDialog* self) {
	auto* vqprogressdialog = dynamic_cast<const VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_SizeHint_IsBase(true);
	return new QSize(vqprogressdialog->sizeHint());
}
}

// Auxiliary method to allow providing re-implementation
void QProgressDialog_OnSizeHint(const QProgressDialog* self, intptr_t slot) {
	auto* vqprogressdialog = dynamic_cast<const VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_SizeHint_Callback(reinterpret_cast<VirtualQProgressDialog::QProgressDialog_SizeHint_Callback>(slot));
}
}

// Derived class handler implementation
void QProgressDialog_ResizeEvent(QProgressDialog* self, QResizeEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
	vqprogressdialog->resizeEvent(event);
	} else {
	self->QProgressDialog::resizeEvent(event);
}
}

// Base class handler implementation
void QProgressDialog_QBaseResizeEvent(QProgressDialog* self, QResizeEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ResizeEvent_IsBase(true);
	vqprogressdialog->resizeEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QProgressDialog_OnResizeEvent(QProgressDialog* self, intptr_t slot) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ResizeEvent_Callback(reinterpret_cast<VirtualQProgressDialog::QProgressDialog_ResizeEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QProgressDialog_CloseEvent(QProgressDialog* self, QCloseEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
	vqprogressdialog->closeEvent(event);
	} else {
	self->QProgressDialog::closeEvent(event);
}
}

// Base class handler implementation
void QProgressDialog_QBaseCloseEvent(QProgressDialog* self, QCloseEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_CloseEvent_IsBase(true);
	vqprogressdialog->closeEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QProgressDialog_OnCloseEvent(QProgressDialog* self, intptr_t slot) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_CloseEvent_Callback(reinterpret_cast<VirtualQProgressDialog::QProgressDialog_CloseEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QProgressDialog_ChangeEvent(QProgressDialog* self, QEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
	vqprogressdialog->changeEvent(event);
	} else {
	self->QProgressDialog::changeEvent(event);
}
}

// Base class handler implementation
void QProgressDialog_QBaseChangeEvent(QProgressDialog* self, QEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ChangeEvent_IsBase(true);
	vqprogressdialog->changeEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QProgressDialog_OnChangeEvent(QProgressDialog* self, intptr_t slot) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ChangeEvent_Callback(reinterpret_cast<VirtualQProgressDialog::QProgressDialog_ChangeEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QProgressDialog_ShowEvent(QProgressDialog* self, QShowEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
	vqprogressdialog->showEvent(event);
	} else {
	self->QProgressDialog::showEvent(event);
}
}

// Base class handler implementation
void QProgressDialog_QBaseShowEvent(QProgressDialog* self, QShowEvent* event) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ShowEvent_IsBase(true);
	vqprogressdialog->showEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QProgressDialog_OnShowEvent(QProgressDialog* self, intptr_t slot) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ShowEvent_Callback(reinterpret_cast<VirtualQProgressDialog::QProgressDialog_ShowEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QProgressDialog_ForceShow(QProgressDialog* self) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
	vqprogressdialog->forceShow();
	} else {
	self->QProgressDialog::forceShow();
}
}

// Base class handler implementation
void QProgressDialog_QBaseForceShow(QProgressDialog* self) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ForceShow_IsBase(true);
	vqprogressdialog->forceShow();
}
}

// Auxiliary method to allow providing re-implementation
void QProgressDialog_OnForceShow(QProgressDialog* self, intptr_t slot) {
	auto* vqprogressdialog = dynamic_cast<VirtualQProgressDialog*>(self);
	if (vqprogressdialog && vqprogressdialog->isVirtualQProgressDialog) {
vqprogressdialog->setQProgressDialog_ForceShow_Callback(reinterpret_cast<VirtualQProgressDialog::QProgressDialog_ForceShow_Callback>(slot));
}
}

void QProgressDialog_Delete(QProgressDialog* self) {
    delete self;
}

